crates/iota-sdk-graphql-client/src/lib.rs

```diff
@@ -1477,12 +1478,43 @@ impl Client {
             let bcs = base64ct::Base64::decode_vec(result.effects.bcs.0.as_str())?;
             let effects: TransactionEffects = bcs::from_bytes(&bcs)?;
 
-            Ok(Some(effects))
+            if wait_for_finalization {
+                self.wait_for_tx_finalization(tx.digest(), None).await?;
+            }
+
+            Ok(effects)
         } else {
-            Ok(None)
+            Err(Error::from_message(
+                Kind::Missing,
+                format!("transaction for digest {}", tx.digest()),
+            ))
         }
     }
 
+    /// Wait for the finalization of a transaction by its digest. An optional
+    /// timeout can be provided, which, if exceeded, will return an error
+    /// (default 60s). Returns the [`TransactionEffects`].
+    pub async fn wait_for_tx_finalization(
+        &self,
+        digest: Digest,
+        timeout: impl Into<Option<Duration>>,
+    ) -> Result<TransactionEffects> {
+        tokio::time::timeout(
+            timeout.into().unwrap_or_else(|| Duration::from_secs(60)),
+            async {
+                loop {
+                    if let Some(effects) = self.transaction_effects(digest).await? {
+                        break Ok(effects);
+                    } else {
+                        tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
```

Comment:
maybe refactor this fn to use `tokio::time:interval` instead to control the polling? I think it's preferred over sleep in these kinds of scenarios.